Pi2BoardPkg/Drivers/DisplayDxe/DisplayDxe.c

```diff
@@ -218,8 +218,22 @@ DisplayBlt(
     )
 {
     UINT8 *VidBuf, *BltBuf, *VidBuf1;
+    UINT32 GopWidth = This->Mode->Info->HorizontalResolution;
+    UINT32 GopHeight = This->Mode->Info->VerticalResolution;
     UINTN i, j;
 
+    if (Width > GopWidth) {
```

Comment:
I don't think it is enough to just bounds-check Width and Height here. Should probably be based on how each BltOperation is using the arguments and bounds-checking within there.
https://github.com/ms-iot/RPi-UEFI/blob/2e4b7dd1fd02b0d2fd7712d37aa31de07e98eba6/MdePkg/Include/Protocol/GraphicsOutput.h#L152-L190

For example, the `EfiBltVideoFill` operation should bounds-check DestinationX, DestinationY, DestinationX+Width, DestinationY+Height. `EfiBltVideoToBltBuffer` should check SourceX, SourceY, SourceX+Width, SourceY+Width.